Polaris registry not support Triple protocol?

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

-->
version 3.0.5
I'm running triple samples by using polaris registry.

**What happened**:
1) modify dubbogo.yml files in `rpc/triple/pb2/go-server/conf` as
```
dubbo:
  registries:
    polarisMesh:
      protocol: polaris
      address: 127.0.0.1:8091
      namespace: dubbogo
  protocols:
    triple:
      name: tri
      port: 20000
  provider:
    services:
      GreeterProvider:
        # interface read from pb file
        interface: ""
```

2) modify dubbogo.yml files in `rpc/triple/pb2/go-client/conf` as
```
dubbo:
  registries:
    polarisMesh:
      protocol: polaris
      address: 127.0.0.1:8091
      namespace: dubbogo
  consumer:
    references:
      GreeterClientImpl:
        protocol: tri
  metrics:
    enable: false
    path: /custom-metrics-path
    port: 9092
```

3) run server command
`DUBBO_GO_CONFIG_PATH=$(pwd)/rpc/triple/pb2/go-server/conf/dubbogo.yml go run rpc/triple/pb2/go-server/cmd/server.go`

output seems right:
```
# command-line-arguments
ld: warning: -no_pie is deprecated when targeting new OS versions
ld: warning: non-standard -pagezero_size is deprecated when targeting macOS 13.0 or later
2023-02-17T20:48:21.061+0800    INFO    config/config_loader_options.go:201     The following profiles are active: default
2023-02-17T20:48:21.062+0800    INFO    config/root_config.go:131       [Config Center] Config center doesn't start
2023-02-17T20:48:21.062+0800    WARN    prometheus/reporter.go:425      new prometheus reporter with error = listen tcp :9090: bind: address already in use
2023-02-17T20:48:21.062+0800    INFO    dubbo3/dubbo3_protocol.go:82    [Triple Protocol] Export service: tri://:20000/grpc.reflection.v1alpha.ServerReflection?accesslog=&app.version=&application=dubbo.io&auth=&bean.name=DubbogoServerReflectionServer&cluster=failover&config.tracing=&environment=&execute.limit=&execute.limit.rejected.handler=&export=true&interface=grpc.reflection.v1alpha.ServerReflection&loadbalance=random&message_size=4&metadata-type=local&module=sample&name=dubbo.io&organization=dubbo-go&owner=dubbo-go&param.sign=&pid=99769&registry.role=3&release=dubbo-golang-3.0.4&retries=&serialization=&service.filter=echo%2Cmetrics%2Ctoken%2Caccesslog%2Ctps%2Cgeneric_service%2Cexecute%2Cpshutdown&side=provider&timestamp=1676638101&tps.limit.interval=&tps.limit.rate=&tps.limit.rejected.handler=&tps.limit.strategy=&tps.limiter=&warmup=
2023-02-17T20:48:21.062+0800    WARN    prometheus/reporter.go:425      new prometheus reporter with error = listen tcp :9090: bind: address already in use
2023-02-17T20:48:21.063+0800    INFO    dubbo3/dubbo3_protocol.go:82    [Triple Protocol] Export service: tri://:20000/org.apache.dubbogo.samples.rpc.triple.pb2.api.Greeter?accesslog=&app.version=&application=dubbo.io&auth=&bean.name=GreeterProvider&cluster=failover&config.tracing=&environment=&execute.limit=&execute.limit.rejected.handler=&export=true&interface=org.apache.dubbogo.samples.rpc.triple.pb2.api.Greeter&loadbalance=random&message_size=4&metadata-type=local&module=sample&name=dubbo.io&organization=dubbo-go&owner=dubbo-go&param.sign=&pid=99769&registry=polaris&registry.role=3&release=dubbo-golang-3.0.4&retries=&serialization=&service.filter=echo%2Cmetrics%2Ctoken%2Caccesslog%2Ctps%2Cgeneric_service%2Cexecute%2Cpshutdown&side=provider&timestamp=1676638101&tps.limit.interval=&tps.limit.rate=&tps.limit.rejected.handler=&tps.limit.strategy=&tps.limiter=&warmup=
2023-02-17T20:48:22.579+0800    WARN    common/rpc_service.go:373       method Resume of mtype func(*health.DubbogoHealthServer) has wrong number of in out parameters 0; needs exactly 1/2
2023-02-17T20:48:22.580+0800    WARN    common/rpc_service.go:373       method SetServingStatus of mtype func(*health.DubbogoHealthServer, string, triple_health_v1.HealthCheckResponse_ServingStatus) has wrong number of in out parameters 0; needs exactly 1/2
2023-02-17T20:48:22.580+0800    WARN    common/rpc_service.go:373       method Shutdown of mtype func(*health.DubbogoHealthServer) has wrong number of in out parameters 0; needs exactly 1/2
2023-02-17T20:48:22.580+0800    INFO    dubbo3/dubbo3_protocol.go:82    [Triple Protocol] Export service: tri://:20000/grpc.health.v1.Health?accesslog=&app.version=&application=dubbo.io&auth=&bean.name=DubbogoHealthServer&cluster=failover&config.tracing=&environment=&execute.limit=&execute.limit.rejected.handler=&export=true&interface=grpc.health.v1.Health&loadbalance=random&message_size=4&metadata-type=local&module=sample&name=dubbo.io&organization=dubbo-go&owner=dubbo-go&param.sign=&pid=99769&registry.role=3&release=dubbo-golang-3.0.4&retries=&serialization=&service.filter=echo%2Cmetrics%2Ctoken%2Caccesslog%2Ctps%2Cgeneric_service%2Cexecute%2Cpshutdown&side=provider&timestamp=1676638102&tps.limit.interval=&tps.limit.rate=&tps.limit.rejected.handler=&tps.limit.strategy=&tps.limiter=&warmup=
2023-02-17T20:48:22.580+0800    INFO    dubbo/dubbo_protocol.go:84      [DUBBO Protocol] Export service: dubbo://:52486/org.apache.dubbo.metadata.MetadataService?accesslog=&app.version=&application=dubbo.io&auth=&bean.name=MetadataService&cluster=&config.tracing=&environment=&execute.limit=&execute.limit.rejected.handler=&export=true&group=dubbo.io&interface=org.apache.dubbo.metadata.MetadataService&loadbalance=&message_size=0&metadata-type=local&module=sample&name=dubbo.io&organization=dubbo-go&owner=dubbo-go&param.sign=&pid=99769&registry.role=3&release=dubbo-golang-3.0.4&retries=&serialization=&service.filter=echo%2Cmetrics%2Ctoken%2Caccesslog%2Ctps%2Cgeneric_service%2Cexecute%2Cpshutdown&side=provider&timestamp=1676638102&tps.limit.interval=&tps.limit.rate=&tps.limit.rejected.handler=&tps.limit.strategy=&tps.limiter=&version=1.0.0&warmup=
2023-02-17T20:48:22.582+0800    INFO    configurable/exporter.go:80     [Metadata Service] The MetadataService exports urls : [dubbo://:52486/org.apache.dubbo.metadata.MetadataService?accesslog=&app.version=&application=dubbo.io&auth=&bean.name=MetadataService&cluster=&config.tracing=&environment=&execute.limit=&execute.limit.rejected.handler=&export=true&group=dubbo.io&interface=org.apache.dubbo.metadata.MetadataService&loadbalance=&message_size=0&metadata-type=local&module=sample&name=dubbo.io&organization=dubbo-go&owner=dubbo-go&param.sign=&pid=99769&registry.role=3&release=dubbo-golang-3.0.4&retries=&serialization=&service.filter=echo%2Cmetrics%2Ctoken%2Caccesslog%2Ctps%2Cgeneric_service%2Cexecute%2Cpshutdown&side=provider&timestamp=1676638102&tps.limit.interval=&tps.limit.rate=&tps.limit.rejected.handler=&tps.limit.strategy=&tps.limiter=&version=1.0.0&warmup=] 
```

and I can see the provider listed in polaris console with name `providers:org.apache.dubbogo.samples.rpc.triple.pb2.api.Greeter` in namespace `dubbogo`

4) run client command
`DUBBO_GO_CONFIG_PATH=$(pwd)/rpc/triple/pb2/go-client/conf/dubbogo.yml go run rpc/triple/pb2/go-client/cmd/client.go`

the output:
```
# command-line-arguments
ld: warning: -no_pie is deprecated when targeting new OS versions
ld: warning: non-standard -pagezero_size is deprecated when targeting macOS 13.0 or later
2023-02-17T20:48:35.825+0800    INFO    config/config_loader_options.go:201     The following profiles are active: default
2023-02-17T20:48:35.826+0800    INFO    config/root_config.go:131       [Config Center] Config center doesn't start
2023-02-17T20:48:36.843+0800    INFO    directory/directory.go:253      [Registry Directory] selector add service url{dubbo://192.168.32.58:20000/org.apache.dubbogo.samples.rpc.triple.pb2.api.Greeter?DUBBOPATH=%2Forg.apache.dubbogo.samples.rpc.triple.pb2.api.Greeter&accesslog=&app.version=&application=dubbo.io&auth=&bean.name=GreeterProvider&cluster=failover&config.tracing=&environment=&execute.limit=&execute.limit.rejected.handler=&export=true&healthstatus=false&interface=org.apache.dubbogo.samples.rpc.triple.pb2.api.Greeter&isolated=false&loadbalance=random&message_size=4&metadata-type=local&module=sample&name=dubbo.io&organization=dubbo-go&owner=dubbo-go&param.sign=&pid=99769&polaris.instanceID=17bcfa89f2143d815b98254c9a818418fc628143&protocol=dubbo&registry=polaris&registry.role=3&release=dubbo-golang-3.0.4&retries=&serialization=&service.filter=echo%2Cmetrics%2Ctoken%2Caccesslog%2Ctps%2Cgeneric_service%2Cexecute%2Cpshutdown&side=provider&timestamp=1676638101&tps.limit.interval=&tps.limit.rate=&tps.limit.rejected.handler=&tps.limit.strategy=&tps.limiter=&warmup=}
2023-02-17T20:48:36.843+0800    INFO    polaris/registry.go:163 update begin, service event: ServiceEvent{Action{add}, Path{dubbo://192.168.32.58:20000/org.apache.dubbogo.samples.rpc.triple.pb2.api.Greeter?DUBBOPATH=%2Forg.apache.dubbogo.samples.rpc.triple.pb2.api.Greeter&accesslog=&app.version=&application=dubbo.io&auth=&bean.name=GreeterProvider&cluster=failover&config.tracing=&environment=&execute.limit=&execute.limit.rejected.handler=&export=true&healthstatus=false&interface=org.apache.dubbogo.samples.rpc.triple.pb2.api.Greeter&isolated=false&loadbalance=random&message_size=4&metadata-type=local&module=sample&name=dubbo.io&organization=dubbo-go&owner=dubbo-go&param.sign=&pid=99769&polaris.instanceID=17bcfa89f2143d815b98254c9a818418fc628143&protocol=dubbo&registry=polaris&registry.role=3&release=dubbo-golang-3.0.4&retries=&serialization=&service.filter=echo%2Cmetrics%2Ctoken%2Caccesslog%2Ctps%2Cgeneric_service%2Cexecute%2Cpshutdown&side=provider&timestamp=1676638101&tps.limit.interval=&tps.limit.rate=&tps.limit.rejected.handler=&tps.limit.strategy=&tps.limiter=&warmup=}, Key{}}
2023-02-17T20:48:36.844+0800    INFO    directory/directory.go:253      [Registry Directory] selector add service url{dubbo://192.168.32.58:20000/org.apache.dubbogo.samples.rpc.triple.pb2.api.Greeter?DUBBOPATH=%2Forg.apache.dubbogo.samples.rpc.triple.pb2.api.Greeter&accesslog=&app.version=&application=dubbo.io&auth=&bean.name=GreeterProvider&cluster=failover&config.tracing=&environment=&execute.limit=&execute.limit.rejected.handler=&export=true&healthstatus=false&interface=org.apache.dubbogo.samples.rpc.triple.pb2.api.Greeter&isolated=false&loadbalance=random&message_size=4&metadata-type=local&module=sample&name=dubbo.io&organization=dubbo-go&owner=dubbo-go&param.sign=&pid=99769&polaris.instanceID=17bcfa89f2143d815b98254c9a818418fc628143&protocol=dubbo&registry=polaris&registry.role=3&release=dubbo-golang-3.0.4&retries=&serialization=&service.filter=echo%2Cmetrics%2Ctoken%2Caccesslog%2Ctps%2Cgeneric_service%2Cexecute%2Cpshutdown&side=provider&timestamp=1676638101&tps.limit.interval=&tps.limit.rate=&tps.limit.rejected.handler=&tps.limit.strategy=&tps.limiter=&warmup=}
2023-02-17T20:48:40.357+0800    ERROR   config/consumer_config.go:158   No provider available of the service org.apache.dubbogo.samples.rpc.triple.pb2.api.Greeter.please check configuration.
dubbo.apache.org/dubbo-go/v3/config.(*ConsumerConfig).Load
        /Users/wumengye/go/pkg/mod/dubbo.apache.org/dubbo-go/v3@v3.0.5/config/consumer_config.go:158
dubbo.apache.org/dubbo-go/v3/config.(*RootConfig).Start.func1
        /Users/wumengye/go/pkg/mod/dubbo.apache.org/dubbo-go/v3@v3.0.5/config/root_config.go:205
sync.(*Once).doSlow
        /usr/local/go/src/sync/once.go:74
sync.(*Once).Do
        /usr/local/go/src/sync/once.go:65
dubbo.apache.org/dubbo-go/v3/config.(*RootConfig).Start
        /Users/wumengye/go/pkg/mod/dubbo.apache.org/dubbo-go/v3@v3.0.5/config/root_config.go:203
dubbo.apache.org/dubbo-go/v3/config.(*RootConfig).Init
        /Users/wumengye/go/pkg/mod/dubbo.apache.org/dubbo-go/v3@v3.0.5/config/root_config.go:198
dubbo.apache.org/dubbo-go/v3/config.Load
        /Users/wumengye/go/pkg/mod/dubbo.apache.org/dubbo-go/v3@v3.0.5/config/config_loader.go:57
main.main
        /Users/wumengye/Documents/GoProjects/dubbo-go-samples/rpc/triple/pb2/go-client/cmd/client.go:45
runtime.main
        /usr/local/go/src/runtime/proc.go:250
panic: No provider available of the service org.apache.dubbogo.samples.rpc.triple.pb2.api.Greeter.please check configuration.

goroutine 1 [running]:
dubbo.apache.org/dubbo-go/v3/config.(*ConsumerConfig).Load(0xc0003f8f00)
        /Users/wumengye/go/pkg/mod/dubbo.apache.org/dubbo-go/v3@v3.0.5/config/consumer_config.go:159 +0x8ac
dubbo.apache.org/dubbo-go/v3/config.(*RootConfig).Start.func1()
        /Users/wumengye/go/pkg/mod/dubbo.apache.org/dubbo-go/v3@v3.0.5/config/root_config.go:205 +0x36
sync.(*Once).doSlow(0x40108e7?, 0xa8?)
        /usr/local/go/src/sync/once.go:74 +0xc2
sync.(*Once).Do(...)
        /usr/local/go/src/sync/once.go:65
dubbo.apache.org/dubbo-go/v3/config.(*RootConfig).Start(0x55b5040?)
        /Users/wumengye/go/pkg/mod/dubbo.apache.org/dubbo-go/v3@v3.0.5/config/root_config.go:203 +0x46
dubbo.apache.org/dubbo-go/v3/config.(*RootConfig).Init(0xc0001b8e70)
        /Users/wumengye/go/pkg/mod/dubbo.apache.org/dubbo-go/v3@v3.0.5/config/root_config.go:198 +0x565
dubbo.apache.org/dubbo-go/v3/config.Load({0x0?, 0x69223f0?, 0xc0000061a0?})
        /Users/wumengye/go/pkg/mod/dubbo.apache.org/dubbo-go/v3@v3.0.5/config/config_loader.go:57 +0xb1
main.main()
        /Users/wumengye/Documents/GoProjects/dubbo-go-samples/rpc/triple/pb2/go-client/cmd/client.go:45 +0x25
exit status 2

```


**What you expected to happen**:
The client should output:
```
2023-02-17T20:50:52.932+0800    INFO    cmd/client.go:54        >>>>> Dubbo-go client is about to call to SayHelloStream
2023-02-17T20:50:52.933+0800    INFO    cmd/client.go:80        Receive 1 user = &User{ID:123456789,Name:hello laurence,Age:18,}

2023-02-17T20:50:52.934+0800    INFO    cmd/client.go:90        Receive 2 user = &User{ID:123456789,Name:hello laurence,Age:19,}

2023-02-17T20:50:52.934+0800    INFO    cmd/client.go:94        >>>>> Dubbo-go client is about to call to SayHello
2023-02-17T20:50:52.935+0800    INFO    cmd/client.go:107       Receive user = &User{ID:12345,Name:Hello laurence,Age:21,}

```


**How to reproduce it (as minimally and precisely as possible)**:
can be reproduced.

**Anything else we need to know?**:

--- a/common/url.go
+++ b/common/url.go
@@ -36,7 +36,9 @@ import (
 	gxset "github.com/dubbogo/gost/container/set"
 
 	"github.com/google/uuid"
+
 	"github.com/jinzhu/copier"
+
 	perrors "github.com/pkg/errors"
 )
 

--- a/config/metric_config.go
+++ b/config/metric_config.go
@@ -19,6 +19,7 @@ package config
 
 import (
 	"github.com/creasty/defaults"
+
 	"github.com/dubbogo/gost/log/logger"
 
 	"github.com/pkg/errors"

--- a/registry/polaris/registry.go
+++ b/registry/polaris/registry.go
@@ -255,12 +255,17 @@ func createRegisterParam(url *common.URL, serviceName string) *api.InstanceRegis
 
 	ver := url.GetParam("version", "")
 
+	protocolVal := protocolForDubboGO
+	if len(url.Protocol) != 0 {
+		protocolVal = url.Protocol
+	}
+
 	req := &api.InstanceRegisterRequest{
 		InstanceRegisterRequest: model.InstanceRegisterRequest{
 			Service:  serviceName,
 			Host:     url.Ip,
 			Port:     port,
-			Protocol: &protocolForDubboGO,
+			Protocol: &protocolVal,
 			Version:  &ver,
 			Metadata: metadata,
 		},

--- a/registry/polaris/registry.go
+++ b/registry/polaris/registry.go
@@ -255,17 +255,12 @@ func createRegisterParam(url *common.URL, serviceName string) *api.InstanceRegis
 
 	ver := url.GetParam("version", "")
 
-	protocolVal := protocolForDubboGO
-	if len(url.Protocol) != 0 {
-		protocolVal = url.Protocol
-	}
-
 	req := &api.InstanceRegisterRequest{
 		InstanceRegisterRequest: model.InstanceRegisterRequest{
 			Service:  serviceName,
 			Host:     url.Ip,
 			Port:     port,
-			Protocol: &protocolVal,
+			Protocol: &url.Protocol,
 			Version:  &ver,
 			Metadata: metadata,
 		},

--- a/registry/polaris/service_discovery.go
+++ b/registry/polaris/service_discovery.go
@@ -317,9 +317,12 @@ func (polaris *polarisServiceDiscovery) String() string {
 
 func convertToRegisterInstance(namespace string, instance registry.ServiceInstance) *api.InstanceRegisterRequest {
 
-	health := instance.IsHealthy()
-	isolate := instance.IsEnable()
-	ttl := 5
+	var (
+		health      = instance.IsHealthy()
+		isolate     = instance.IsEnable()
+		ttl         = 5
+		protocolVal = protocolForDubboGO
+	)
 
 	return &api.InstanceRegisterRequest{
 		InstanceRegisterRequest: model.InstanceRegisterRequest{
@@ -327,7 +330,7 @@ func convertToRegisterInstance(namespace string, instance registry.ServiceInstan
 			Namespace: namespace,
 			Host:      instance.GetHost(),
 			Port:      instance.GetPort(),
-			Protocol:  &protocolForDubboGO,
+			Protocol:  &protocolVal,
 			Metadata:  instance.GetMetadata(),
 			Healthy:   &health,
 			Isolate:   &isolate,

--- a/registry/polaris/service_discovery.go
+++ b/registry/polaris/service_discovery.go
@@ -321,7 +321,7 @@ func convertToRegisterInstance(namespace string, instance registry.ServiceInstan
 		health      = instance.IsHealthy()
 		isolate     = instance.IsEnable()
 		ttl         = 5
-		protocolVal = protocolForDubboGO
+		protocolVal = string(constant.Dubbo)
 	)
 
 	return &api.InstanceRegisterRequest{

--- a/registry/polaris/utils.go
+++ b/registry/polaris/utils.go
@@ -30,10 +30,6 @@ import (
 	"dubbo.apache.org/dubbo-go/v3/registry"
 )
 
-var (
-	protocolForDubboGO string = "dubbo"
-)
-
 type PolarisInstanceInfo struct {
 	instance registry.ServiceInstance
 	url      *common.URL

--- a/remoting/zookeeper/listener.go
+++ b/remoting/zookeeper/listener.go
@@ -23,13 +23,18 @@ import (
 	"sync"
 	"time"
 )
+
 import (
 	"github.com/dubbogo/go-zookeeper/zk"
+
 	gxzookeeper "github.com/dubbogo/gost/database/kv/zk"
 	"github.com/dubbogo/gost/log/logger"
+
 	perrors "github.com/pkg/errors"
+
 	uatomic "go.uber.org/atomic"
 )
+
 import (
 	"dubbo.apache.org/dubbo-go/v3/common"
 	"dubbo.apache.org/dubbo-go/v3/common/constant"

